### PR TITLE
Trusted Environment Variables

### PR DESCRIPTION
**Pull request by jeremytwfortune**
## Description of Changes

The aim of this PR is to streamline creation of the API clients. In main at the moment, we're:

1. Requiring that the user specify their API key via arguments. Doing this appropriately usually means loading your environment and then immediately re-referencing it:

    ```python
    import os
    from orchestrate import OrchestrateApi

    os.environ["ORCHESTRATE_API_KEY"] = "key"
    api = OrchestrateApi(api_key=os.environ["ORCHESTRATE_API_KEY"])
    ```

    This was initially done to keep opinionated environment variables out of the mix, but it's a bit of a pain to use. Instead, this PR uses `ORCHESTRATE_API_KEY` if it is found in the environment and no `api_key` is provided.

    ```python
    import os
    from orchestrate import OrchestrateApi
    
    os.environ["ORCHESTRATE_API_KEY"] = "key"
    api = OrchestrateApi()
    ````
2. The clients allowed for the user to specify a base URL and additional headers, but this is almost certainly useless to end users. This PR removes these options, loading them from the environment. This is breaking, certainly, but if anyone is currently using the functionality, it's basically impossible that they're using it to do anything useful.

## Security

**REMINDER: All file contents are public.**

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc. Of particular note: No temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] My changes do not introduce any security risks, or any such risks have been properly mitigated.

This, ideally, should make it marginally more difficult for users to leave their API keys around in scripts. There is some chance that the user does not change their API key in the environment, executing Orchesrate API calls with the wrong key. While not ideal, it isn't inappropriately leaking data in some form. Encouraging env var use is, on the whole, a better option.

## Reviewers

- [x] I have assigned the appropriate reviewer(s).

@CareEvolution/api-docs : The breaking part of this PR isn't documented to my knowledge, but it might be useful to call out the feature somewhere.

### CODE REVIEW

File: typescript/tests/httpHandler.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      vi.stubEnv("ORCHESTRATE_BASE_URL", "https://env.example.com");
+
+      createHttpHandler(undefined);
+
+      expect(HttpHandler).toHaveBeenCalledWith("https://env.example.com", {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      });
+    } finally {
+      vi.unstubAllEnvs();
```
Comment (kneumei) [inline typescript/tests/httpHandler.test.ts:29]: I think it's more typical to run this kind of thing in a `beforeEach`

File: typescript/src/httpHandlerFactory.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+function getPriorityBaseUrl(): string {
+  if (process.env.ORCHESTRATE_BASE_URL) {
+    return process.env.ORCHESTRATE_BASE_URL;
+  }
+  return "https://api.careevolutionapi.com";
+}
+
+function getPriorityApiKey(apiKey: string | undefined): string | undefined {
+  if (apiKey) {
```
Comment (kneumei) [inline typescript/src/httpHandlerFactory.ts:11]: I think this whole method could be 

`return apiKey || process.env.ORCHESTRATE_API_KEY`

File: typescript/src/httpHandlerFactory.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { HttpHandler, IHttpHandler } from "./httpHandler";
+
+function getPriorityBaseUrl(): string {
+  if (process.env.ORCHESTRATE_BASE_URL) {
```
Comment (kneumei) [inline typescript/src/httpHandlerFactory.ts:4]: `return process.env.ORCHESTRATE_BASE_URL || "https://api.careevolutionapi.com"`

File: typescript/src/httpHandlerFactory.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+export function createHttpHandler(
+  apiKey: string | undefined,
+): IHttpHandler {
+  const baseUrl = getPriorityBaseUrl();
+  const additionalHeaders = getAdditionalHeaders();
+
+  const defaultHeaders = {
+    ...(additionalHeaders ?? {}),
+    ...{
```
Comment (kneumei) [inline typescript/src/httpHandlerFactory.ts:35]: I don't think you need the spread operator here. 

Also would prefer declaring the type of the object over casting. You get a little more type safety that way

```
  const defaultHeaders: { [key: string]: string } = {
      "Content-Type": "application/json",
      Accept: "application/json",
      ...additionalHeaders,
    
  }
```

**Review by kneumei — APPROVED** [review]
Just took a look over the typescript stuff. Some minor style comments, but everything looks fine.

**Review by jwyderko — APPROVED** [review]

**Review by kneumei — APPROVED** [review]

**Pull request merged**